src/WebJobs.Script/Content/Script/functions.js

```diff
@@ -49,7 +51,15 @@ function createFunction(f) {
         });
         context.log = log;
 
+        if (typeof func === "string") {
+            systemLog("Requiring user function script.");
+            func = require(func);
+        }
+
+        func = getEntryPoint(func, context);
+
         context.done = function (err, result) {
+            systemLog("context.done called.");
             if (context._done) {
                 if (context._promise) {
                     context.log("Error: Choose either to return a promise or call 'done'.  Do not use both in your script.");
```

Comment:
Exactly as you wrote, because of the way that javascript handles scoping. The context parameter is available for use within the callback.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Closures